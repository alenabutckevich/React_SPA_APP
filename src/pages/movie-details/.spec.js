import React from 'react';
import toJson from 'enzyme-to-json';
import MovieDetails from './movie-details';

const setup = propOverrides => {
    const props = Object.assign({
        director: 'director name'
    }, propOverrides);

    const wrapper = shallow(<MovieDetails {...props}/>);

    return {
        wrapper,
        props
    }
}

describe('<MovieDetails />', () => {

    it('render movie details', () => {
        const { wrapper } = setup();
        const panel = shallow(wrapper.props().panel);

        expect(panel.find('.director-panel').text()).toEqual('Films by director name');
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})