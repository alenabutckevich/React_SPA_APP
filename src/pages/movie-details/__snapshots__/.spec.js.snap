// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieDetails /> render movie details 1`] = `
<Layout
  headerContent={
    <MovieDetailsHeader
      cast=""
      director="director name"
      history={undefined}
      movie={undefined}
      searchQuery={undefined}
    />
  }
  panel={
    <span
      className="director-panel"
    >
      Films by 
      director name
    </span>
  }
/>
`;
